projects/OG-Analytics/src/test/java/com/opengamma/analytics/financial/interestrate/future/provider/InterestRateFutureOptionMarginNormalSmileMethodTest.java

```diff
@@ -172,4 +179,203 @@ public void presentValueNormalSensitivity() {
+        MULTICURVES, surfacedw, EURIBOR3M);
+    double priceVolUp = METHOD_SECURITY_OPTION_NORMAL.price(OPTION_ERU2, normalUp);
+    double priceVolDw = METHOD_SECURITY_OPTION_NORMAL.price(OPTION_ERU2, normalDw);
+    double expectedVega = 0.5 * (priceVolUp - priceVolDw) / eps;
+    assertEquals("Future option with volatilities, Greeks", expectedVega, computedVega, eps);
+
+    double computedGamma = METHOD_SECURITY_OPTION_NORMAL.priceGamma(OPTION_ERU2, NORMAL_MULTICURVES);
+    double deltaFutUp = METHOD_SECURITY_OPTION_NORMAL.priceDeltaFromFuturePrice(OPTION_ERU2,
+        NORMAL_MULTICURVES, priceFuture + eps);
+    double deltaFutDw = METHOD_SECURITY_OPTION_NORMAL.priceDeltaFromFuturePrice(OPTION_ERU2,
+        NORMAL_MULTICURVES, priceFuture - eps);
+    double expectedGamma = 0.5 * (deltaFutUp - deltaFutDw) / eps;
+    assertEquals("Future option with volatilities, Greeks", expectedGamma, computedGamma, eps);
+
+    double computedTheta = METHOD_SECURITY_OPTION_NORMAL.priceTheta(OPTION_ERU2, NORMAL_MULTICURVES);
+    double time = OPTION_ERU2.getExpirationTime();
+    InterestRateFutureOptionMarginSecurity OptionUp = new InterestRateFutureOptionMarginSecurity(ERU2, time + eps,
+        STRIKE, IS_CALL);
+    InterestRateFutureOptionMarginSecurity OptionDw = new InterestRateFutureOptionMarginSecurity(ERU2, time - eps,
+        STRIKE, IS_CALL);
+    double priceTimeUp = METHOD_SECURITY_OPTION_NORMAL.priceFromFuturePrice(OptionUp, NORMAL_MULTICURVES, priceFuture);
+    double priceTimeDw = METHOD_SECURITY_OPTION_NORMAL.priceFromFuturePrice(OptionDw, NORMAL_MULTICURVES, priceFuture);
+    double volatilityUp = NORMAL_MULTICURVES.getVolatility(time + eps, 0.0, STRIKE, priceFuture);
+    double volatilityDw = NORMAL_MULTICURVES.getVolatility(time - eps, 0.0, STRIKE, priceFuture);
+    double expectedTheta = -0.5 * (priceTimeUp - priceTimeDw) / eps + 0.5 * computedVega *
+        (volatilityUp - volatilityDw) / eps;
+    assertEquals("Future option with volatilities, Greeks", expectedTheta, computedTheta, eps);
+  }
+
+  /* 
+   * Tests below use NormalSTIRFuturesExpSimpleMoneynessProviderDiscount
+   */
+  private static final InterpolatedDoublesSurface NORMAL_PARAMETERS_MONEYNESS = NormalDataSets
+      .createNormalSurfaceFuturesPricesSimpleMoneyness();
+  private static final NormalSTIRFuturesExpSimpleMoneynessProviderDiscount NORMAL_MULTICURVES_MONEYNESS = new NormalSTIRFuturesExpSimpleMoneynessProviderDiscount(
+      MULTICURVES, NORMAL_PARAMETERS_MONEYNESS, EURIBOR3M);
+
+  /**
+   * Test the option price from the future price.
+   */
+  @Test
+  public void priceMoneyness() {
+    double expiry = OPTION_ERU2.getExpirationTime();
+    EuropeanVanillaOption option = new EuropeanVanillaOption(STRIKE, expiry, IS_CALL);
+    double priceFuture = METHOD_FUTURES.price(ERU2, MULTICURVES);
+    double volatility = NORMAL_PARAMETERS_MONEYNESS.getZValue(expiry, STRIKE);
```

Comment:
Passing the STRIKE instead of moneyness. The test pass probably because the smile is extrapolated flat and there are very few points in the dataset (only 0.99 and 0.995 strikes). The number of strikes in the volatility surface should be increased to cover at least the strike of the option tested. The volatilities in the surface should all be different (to avoid having too many volatility request returning the same number).